OpenIDConnectClient.php

```diff
@@ -417,7 +417,7 @@ public function getRedirectURL() {
               ?: @$_SERVER['SERVER_ADDR'];
 
         // Don't include port if it's 80 or 443 and the protocol matches
-        $port = (443 == $port) || (80 == $port) ? '' : ':' . $port;
+        $port = ($protocol === 'https' && $port === 443) || ($protocol === 'http' && $port === 80) ? '' : ':' . $port;
```

Comment:
The comment says the protocol has to match - should I keep things sane and remove the comment too? :)